rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -790,132 +827,45 @@ def test_accounting_average_cost_basis(accountant):
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(-356.25)
-    assert cost_basis.missing_acquisitions == [
-        MissingAcquisition(
-            asset=asset,
-            time=0,
-            found_amount=FVal(2),
-            missing_amount=FVal(1.5),
-        ),
-    ]
+    manager = cost_basis.get_events(A_ETH).acquisitions_manager
+
+    # Check 2 simple cases - a spend with positive pnl and a spend with negative pnl
+    add_acquisition(pot, amount=FVal(2), price=FVal(10))  # Buy 2 ETH for $10  total acb: $20
+    current_total_acb = events[0].price * events[0].taxable_amount
+    current_amount = events[0].taxable_amount  # 2
+    assert manager.current_total_acb == current_total_acb == FVal(20)
+    add_spend(pot, amount=FVal(1), price=FVal(15))  # Sell 1 ETH for $15  pnl: 1 * (15 - 10) = $5
+    assert events[1].pnl.taxable == events[1].taxable_amount * (events[1].price - current_total_acb / current_amount) == FVal(5)  # noqa: E501
+    current_total_acb *= (current_amount - events[1].taxable_amount) / current_amount  # total acb: 20 * (2 - 1) / 2 = $10  # noqa: E501
+    current_amount -= events[1].taxable_amount  # 1
+    add_acquisition(pot, amount=FVal(1), price=FVal(30))  # Buy 1 ETH for $30  total acb: 10 + 1 * 30 = $40  # noqa: E501
+    current_total_acb += events[2].price * events[2].taxable_amount
+    current_amount += events[2].taxable_amount  # 2
+    assert manager.current_total_acb == current_total_acb == FVal(40)
+    add_spend(pot, amount=FVal(1.5), price=FVal(40))  # Sell 1.5 ETH for $40  pnl: 1.5 * (40 - 20) = $30  # noqa: E501
+    assert events[3].pnl.taxable == events[3].taxable_amount * (events[3].price - current_total_acb / current_amount) == FVal(30)  # noqa: E501
+    current_total_acb *= (current_amount - events[3].taxable_amount) / current_amount  # total acb: 40 * (2 - 1.5) / 2 = $10  # noqa: E501
+    current_amount -= events[3].taxable_amount  # 0.5
+
+    # Now check that if acquisitions don't change the cost basis, everything remains correct.
+    add_acquisition(pot, amount=FVal(3), price=FVal(20))  # Buy 3 ETH for $20  total acb: 10 + 3 * 20 = $70  # noqa: E501
+    current_total_acb += events[4].price * events[4].taxable_amount
+    current_amount += events[4].taxable_amount  # 3.5
+    assert manager.current_total_acb == current_total_acb == FVal(70)
+    add_spend(pot, amount=FVal(2), price=FVal(30))  # Sell 2 ETH for $30  pnl: 2 * (30 - 20) = $20
+    assert events[5].pnl.taxable == events[5].taxable_amount * (events[5].price - current_total_acb / current_amount) == FVal(20)  # noqa: E501
+    current_total_acb *= (current_amount - events[5].taxable_amount) / current_amount  # total acb: 70 * (3.5 - 2) / 3.5 = $30 # noqa: E501
+    current_amount -= events[5].taxable_amount  # 1.5
+
+    # Check that spending up the entire remaining amount works.
+    add_spend(pot, amount=FVal(1.5), price=FVal(10))  # Sell 1.5 ETH for $10  pnl: 1.5 * (10 - 20) = $15  # noqa: E501
```

Comment:
This is wrong. Pnl here is negative.
```suggestion
    add_spend(pot, amount=FVal(1.5), price=FVal(10))  # Sell 1.5 ETH for $10  pnl: 1.5*10 - (30/1.5)*1.5 = -$15 # noqa: E501
```